### DISCUSSION TITLE
Can the MQTT target be changed programmatically?

### DISCUSSION THREAD

**Discussion by wsw70** [krahabb/meross_lan#271] [Q&A]
This question is not strictly related to `meross_lan`, but since there has been so much work done on the protocol, etc. I am taking my chance here :) Any better places to ask this question are very much welcome!

Everything works. The flashed Meross devices, the visibility in HA, automations. So it is the best time to break things and I realized that a lot of my settings on the devices point to the MQTT on `192.168.10.2`  instead of  `mqtt.local` (or whatever FQDN). This makes my life difficult because I would like to use 192.168.10.2 for something else.

Is there a way to **programmatically** (i.e. not by reflashing) to swap  `192.168.10.2` with `mqtt.local` in the device configuration?

**Comment by krahabb** [comment]
Hello @wsw70,
Do you mean issuing an `"Appliance.Config.Key"` to the device to just set the server name?
If I remember correctly that namespace is rejected unless the device is in pairing mode (I did some experiments in the beginning of `meross_lan` development)
I could try that or 'even better' you could use the `meross_lan.request` service from the HA developers tools to try issue the request and see what happens

PS:
Sorry to not be much supportive with testing but I'm in a crazy development phase and I'd rather not start new 'experiments' at the moment. If it works I would be happy to eventually integrate the feature to allow changing the configured mqtt host for any device ;)

**Reply by wsw70** [reply]
> Do you mean issuing an `"Appliance.Config.Key"` to the device to just set the server name?

I will have to check the name but that was the name or IP of the MQTT broker that was used during pairing

> If I remember correctly that namespace is rejected unless the device is in pairing mode (I did some experiments in the beginning of `meross_lan` development)
> I could try that or 'even better' you could use the `meross_lan.request` service from the HA developers tools to try issue the request and see what happens

Thanks, I will try that tonight when I set up a spare device and indeed see what happens :)

> Sorry to not be much supportive with testing but I'm in a crazy development phase and I'd rather not start new 'experiments' at the moment. If it works I would be happy to eventually integrate the feature to allow changing the configured mqtt host for any device ;)

You must not be serious - I thew a question hoping to get something and you provided me a full path. Thank you very much! Good luck with the coding - I am there too, trying to leverage a db framework in Go for the home chores system (who has to get the trash out, ...)

PS. I was back in Livigno this winter to ski - some 30 years after my last time.
